medium/profiles/services.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any, BinaryIO, Dict, Optional
+
+from django.core.exceptions import ValidationError as DjangoValidationError
+from rest_framework.exceptions import ValidationError as DRFValidationError
+
+from medium.common.services import model_update
+from medium.profiles.models import Profile
+from medium.users.models import User
+
+
+class ProfileService:
+    def __init__(self, user_id: int, profile_pic: BinaryIO, header_pic: BinaryIO):
+        self.user_id = user_id
+        self.profile_pic = profile_pic
+        self.header_pic = header_pic
+
+    def create_profile(
+        self,
+        about_text: str,
+        **optional_fields: Optional[Dict[str, Any]],
+    ) -> Profile:
+        try:
+            user: User = User.objects.get(id=self.user_id)
+            profile = Profile(
+                user=user,
+                about_text=about_text,
+                profile_pic=self.profile_pic,
+                header_pic=self.header_pic,
+            )
+
+            for field in optional_fields:
+                setattr(profile, field, optional_fields[field])
+
+            profile.full_clean()
+            profile.save()
+            return profile
+
+        except DjangoValidationError as e:
+            raise DRFValidationError(e.messages)
+
+    def update_profile(self, data: Dict[str, Any]) -> Profile:
+        try:
+            profile = Profile.objects.get(user_id=self.user_id)
+            profile_fields = [
+                "short_bio",
+                "about_text",
+                "profile_pic",
+                "profile_views",
+                "accent_color",
+                "background_color",
+                "header_pic",
+            ]
```

Comment:
I think no need to hard code these fields in the service for two reasons
- If the input data has fields that aren't defined in the model class, the model will ignore it and will not save it into the database.
- If anyone in the future tries to update the model and forgets to update the service this will introduce a bug. and I think it will be a better design if we decouple the two classes.

Reply:
I agree, I was trying to use the generic `model_update` in the style guide. But yeah It causes a coupling problem and magic values in the code as you mentioned. 

Handling it manually seems to be the better option, I'll be changing that

Reply:
what's your purposed solution could you provide an example?

Reply:
Hardcoding the attributes is somehow brittle, we can hardcode the args and manually check them.

The coupling is still there and it's more verbose. However it's more caller-friendly, the caller can see the allowed fields in the signature, and it's safe

Reply:
> what's your purposed solution could you provide an example? @hazemessam

It's already exist, please check the [`update_user`](https://github.com/OsamaRagab520/medium/blob/d272cb569bfbb51d1e3b3aee13767a000bacc468/medium/users/services.py#L36-L58) service.